### ISSUE TITLE
Support precompiling NIfs

### ISSUE THREAD

**Issue by maennchen** [E-xyza/zigler#479]
It would be awesome if it was possible to precompile (and possibly cross-compile for different architectures) the Nifs for published libraries. This would make the installation by the user of the library a lot and removes the requirement to install Zig.

This has already been done by other Nif providers like `rustler`:
https://github.com/philss/rustler_precompiled

It usually works by building the nifs and adding the object files to a GitHub release.

When a user installs the library from Hex, the library looks if the files exist locally and downloads them from the release if not.

**Comment by ityonemo** [comment]
yep.  This is on the todo list in VERSIONS.md  I will try to do it for 0.14 (thought I had an issue up for it but i don't see it anymore)

**Comment by ityonemo** [comment]
it is going to be difficult to support this in this release!  But i will either solve it immediately after the release or make it a high priority in 0.15

**Comment by ityonemo** [comment]
completed 2f6e770
